lib/OpenQA/Schema/Result/Jobs.pm

```diff
@@ -2149,14 +2149,16 @@ sub dependencies {
     my $is_final   = grep { $_ eq $state } FINAL_STATES;
     my $parents_ok = $result ne SKIPPED && $result ne PARALLEL_FAILED && $result ne PARALLEL_RESTARTED;
 
-    my $jp = $self->parents;
-    while (my $s = $jp->next) {
+    $parents_list ||= [$self->parents->all];
+    for my $s (@$parents_list) {
         push(@{$parents{$s->to_string}}, $s->parent_job_id);
         $has_parents = 1;
-        $parents_ok  = $s->parent->is_ok if $is_final && $parents_ok;
+        $parents_ok
+          = $self->result_source->schema->resultset('Jobs')->find($s->parent_job_id, {select => ['result']})->is_ok
```

Comment:
as we already wrapping over the line I suggest to avoid the train wreck with

```suggestion
        my $jobs = $self->result_source->schema->resultset('Jobs');
        $parents_ok = $jobs->find($s->parent_job_id, {select => ['result']})->is_ok
```

Reply:
done